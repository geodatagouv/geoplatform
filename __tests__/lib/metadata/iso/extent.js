const {getCoveredTerritories, getConsolidatedExtent} = require('../../../../lib/metadata/iso/extent')

describe('lib.metadata.iso.extent', () => {
  describe('getCoveredTerritories()', () => {
    it('should return an empty array if not defined', () => {
      expect(getCoveredTerritories({})).toEqual([])
    })

    it('should return an empty array if not geographicElement is defined', () => {
      expect(getCoveredTerritories({
        identificationInfo: {
          extent: [{}, {}]
        }
      })).toEqual([])
    })

    it('should match fr:region:93', () => {
      expect(getCoveredTerritories({
        identificationInfo: {
          extent: [{
            geographicElement: {
              westBoundLongitude: 4.21911199683101,
              eastBoundLongitude: 7.818162622198166,
              southBoundLatitude: 43.0377552055724,
              northBoundLatitude: 45.0667722061971
            }
          }]
        }
      })).toEqual([
        'fr:region:93'
      ])
    })
  })

  describe('getConsolidatedExtent()', () => {
    it('should return null if not defined', () => {
      expect(getConsolidatedExtent({})).toBeNull()
    })

    it('should return null if not geographicElement is defined', () => {
      expect(getConsolidatedExtent({
        identificationInfo: {
          extent: [{}, {}]
        }
      })).toBeNull()
    })

    it('should return a GeoJSON polygon of the extent', () => {
      expect(getConsolidatedExtent({
        identificationInfo: {
          extent: [{
            geographicElement: {
              westBoundLongitude: 4.21911199683101,
              eastBoundLongitude: 7.818162622198166,
              southBoundLatitude: 43.0377552055724,
              northBoundLatitude: 45.0667722061971
            }
          }]
        }
      })).toMatchSnapshot()
    })
  })
})
