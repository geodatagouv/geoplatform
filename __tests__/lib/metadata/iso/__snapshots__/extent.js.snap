// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.metadata.iso.extent getConsolidatedExtent() should return a GeoJSON polygon of the extent 1`] = `
Object {
  "coordinates": Array [
    Array [
      Array [
        4.21911199683101,
        43.0377552055724,
      ],
      Array [
        7.818162622198166,
        43.0377552055724,
      ],
      Array [
        7.818162622198166,
        45.0667722061971,
      ],
      Array [
        4.21911199683101,
        45.0667722061971,
      ],
      Array [
        4.21911199683101,
        43.0377552055724,
      ],
    ],
  ],
  "type": "Polygon",
}
`;
